src/testRunner/unittests/services/preProcessFile.ts

```diff
@@ -88,6 +88,72 @@ describe("unittests:: services:: PreProcessFile:", () => {
                 });
         });
 
+        // https://github.com/microsoft/TypeScript/issues/33680#issue-500399194
+        it("Correctly return dynamically imported files using string literals", () => {
+            test("import(`r1.ts`);",
+                /*readImportFile*/ true,
+                /*detectJavaScriptImports*/ false,
+                {
+                    referencedFiles: <ts.FileReference[]>[],
+                    typeReferenceDirectives: [],
+                    libReferenceDirectives: [],
+                    importedFiles: [{ fileName: "r1.ts", pos: 7, end: 12 }],
+                    ambientExternalModules: undefined,
+                    isLibFile: false
+                });
+        });
+
+        // https://github.com/microsoft/TypeScript/issues/33680#issue-500399194
+        it("Correctly return required files using string literals in JS", () => {
+            test("require(`r1.ts`);",
+                /*readImportFile*/ true,
+                /*detectJavaScriptImports*/ true,
+                {
+                    referencedFiles: <ts.FileReference[]>[],
+                    typeReferenceDirectives: [],
+                    libReferenceDirectives: [],
+                    importedFiles: [{ fileName: "r1.ts", pos: 8, end: 13 }],
+                    ambientExternalModules: undefined,
+                    isLibFile: false
+                });
+        });
+
+        // https://github.com/microsoft/TypeScript/issues/33680#issue-500399194
+        it("Correctly return required files using string literals", () => {
+            test("import i = require(`r1.ts`);",
+                /*readImportFile*/ true,
+                /*detectJavaScriptImports*/ false,
+                {
+                    referencedFiles: <ts.FileReference[]>[],
+                    typeReferenceDirectives: [],
+                    libReferenceDirectives: [],
+                    importedFiles: [{ fileName: "r1.ts", pos: 19, end: 24 }],
+                    ambientExternalModules: undefined,
+                    isLibFile: false
+                });
+        });
+
+        // https://github.com/microsoft/TypeScript/issues/33680#issue-500399194
+        it("Correctly return static imports using string literals", () => {
+            test("import d from `r1.ts`; import d, {a, b as B} from `r2.ts`; import d, * as NS from `r3.ts`; export {a, b as B} from `r4.ts`; export * from `r5.ts`;",

```

Comment:
This is not valid JS/TS (https://github.com/microsoft/TypeScript/issues/29318) should it be parsed here?

Reply:
If this is the only thing blocking the rest of the PR I am happy to roll the change back and open it in a separate request where we can then have the discussion?

Reply:
Then how's different from #33688 with previous roll back?

Reply:
logically they are very similar - I don't completely understand the reason for the introduction of https://github.com/microsoft/TypeScript/pull/33688/files#diff-23f96e78fc240e5f0198a9ffef3a69e4R227; however, I believe the way I structured the tests, updated the comments and implemented the union for literal likes is more in line with the rest of the codebase. When I opened the PR unfortunately I didn't see #33688, otherwise I could have based it on top.

Reply:
Because ``import mod = require(`mod`);`` is error in TS `String literal expected.`
They have same validation as ``import mod from `mod`;``